nautobot/dcim/models/devices.py

```diff
@@ -51,7 +52,7 @@ class Manufacturer(OrganizationalModel):
     """
 
     name = models.CharField(max_length=100, unique=True)
-    slug = models.SlugField(max_length=100, unique=True)
+    slug = AutoSlugField(populate_from="name", max_length=100, unique=True, editable=True, overwrite_on_add=False)
```

Comment:
What do you think about creating our own custom subclass of `AutoSlugField` in `nautobot.utilities.fields` that would have all of these default values as presets? Would reduce the amount of code we have to repeat all over the place and make it easier to ensure that we're defining all of our slug fields consistently across the board.


```python
class AutoSlugField(django_extensions.db.fields.AutoSlugField):

    def __init__(self, *args, **kwargs):
        kwargs.setdefault("max_length", 100)
        kwargs.setdefault("unique", True)
        kwargs.setdefault("editable", True)
        kwargs.setdefault("overwrite_on_add", False)
        super().__init__(*args, **kwargs)
```

Reply:
I like that, and will make these changes shortly.